### ISSUE TITLE
Shelly switches dat toggle icon

### ISSUE THREAD

**Issue by Mariusthvdb** [home-assistant/frontend#19784]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

as title.
Somehow the Shelly integration switches no longer change their icon and remain always using the 'on icon.

before these changed correctly according to either domain, or device_class.

### Describe the behavior you expected

toggle the icon according to state for domain Switch

### Steps to reproduce the issue

1. toggle Shelly integration switch
2.
3.
...

### What version of Home Assistant Core has the issue?

2024.2.1

### What was the last working version of Home Assistant Core?

2023?

### In which browser are you experiencing the issue with?

any

### Additional information

Off (state_color is correct)

<img width="474" alt="Scherm­afbeelding 2024-02-12 om 14 08 20" src="https://github.com/home-assistant/frontend/assets/33354141/04162e51-db0d-4301-8811-628a1d290cd5">

On (icon and color are correct)

<img width="482" alt="Scherm­afbeelding 2024-02-12 om 14 08 35" src="https://github.com/home-assistant/frontend/assets/33354141/94eaa7ae-6766-4ecf-9156-5f458193b867">

**Comment by karwosts** [comment]
Just for reference, here's the icon spec in the backend. Believe this says for switch with no class, always use mdi:toggle-switch-variant, and for switch-class switches, `off` state uses mdi:toggle-switch-variant-off. 

So that tracks with what you're seeing. 

https://github.com/home-assistant/core/blob/bc0a78d0697fed39c5b499103f5c728abb3a4f3c/homeassistant/components/switch/icons.json#L2-L11

```
  "entity_component": {
    "_": {
      "default": "mdi:toggle-switch-variant"
    },
    "switch": {
      "default": "mdi:toggle-switch-variant",
      "state": {
        "off": "mdi:toggle-switch-variant-off"
      }
    },
 ```

**Comment by Mariusthvdb** [comment]
yep, thats the new version of my earlier PR to the device_class switch alright.

so the question is why were those device_classes lost?

**Comment by piitaya** [comment]
With the `2024.2`, icons are migrated to back-end but we have the same rules so it can be the cause of the issue.

Here's the front-end code before `2024.2`.

```ts
    case "switch":
      switch (stateObj?.attributes.device_class) {
        case "outlet":
          return compareState === "on" ? mdiPowerPlug : mdiPowerPlugOff;
        case "switch":
          return compareState === "on"
            ? mdiToggleSwitchVariant
            : mdiToggleSwitchVariantOff;
        default:
          return mdiToggleSwitchVariant;
      }
```

I think we can use the same icon as switch `device class` for entities without device class but I'm very interested in the cause of this regression. 
I move the issue to core repository as core rules are handled by core now.
